tests/worktree/worktree.c

```diff
@@ -228,6 +228,30 @@ void test_worktree_worktree__init(void)
 	git_repository_free(repo);
 }
 
+void test_worktree_worktree__add_from_bare(void)
+{
+	git_worktree *wt;
+	git_repository *repo, *wtrepo;
+	git_buf path = GIT_BUF_INIT;
+
+	repo = cl_git_sandbox_init("testrepo.git");
+
+	cl_assert_equal_i(1, git_repository_is_bare(repo));
+	cl_assert_equal_i(0, git_repository_is_worktree(repo));
```

Comment:
As you also modified `git_repository__ensure_not_bare` you should also test that function!

Reply:
But when you test that function any semantic change will be visible in the future.

Reply:
Also `git_repository__ensure_not_bare` is used by other functions, e.g. adding files to index, which was the initial reason for this fix.  So, therefore this should be tested.